File: ComSSA/revisionsessions/ise/BadCode.py
count = 0
conv_num = 0
isNegative = False


def main(args):
    count = 0
    print("How many conversions would you like?")
    conv_num = int(input())
    while conv_num < 0:
        input("Please only enter a positive number")
        conv_num = int(input())


    while conv_num > 0:
        choice = int(input("inches(1) or cm(2)?"))
        while choice < 1 or choice > 2:
            input("Please only enter a between 1 and 2")
            choice = int(input())


        if choice == 1: 
            isInches = True
        else:
            isInches = False

        input_val = get_user_double(isInches)
        if input_val < 0:
            isNegative = True
        else:
            isNegative = False

        if isNegative:
            print("Please only enter a positive number")
        else:
            if isInches:
                print("Conversion is:", inches_to_meters(input_val))
            else:
                print("Conversion is:", cm_to_meters_or_inches(input_val, isInches))

        add_to_count()
        conv_num -= 1


    print("Completed", count, "conversions")


def get_user_double(isInches):
    if isInches:
        input_val = float(input("Please enter your inches to convert"))
        while input_val < 0:
            input_val = float(input("Please only enter a positive number"))
    else:
        input_val = float(input("Please enter your cm to convert"))
        while input_val < 0:
            input_val = float(input("Please only enter a positive number"))
    return input_val


def add_to_count():
    count += 1


def inches_to_meters(inches):
    if inches > 0 or inches <= 0:
        print("Converting now")
    count += 1
    conv_num -= 1
    return inches / 39.37


def cm_to_meters_or_inches(cm, to_inches):
    if to_inches:
        output = 39.37 * cm
    else:
        output = cm / 100
    count += 1
    conv_num -= 1
    return output

File: ComSSA/revisionsessions/ise/BadCode.java
package revisionsessions.ise;

import java.util.Scanner;


public class BadCode
{
   public static int count;
   public static boolean isNegative;
   public static Scanner sc;
   public static int convNum;


    public static void main( String[] args )
    {
        sc = new Scanner( System.in );
        count = 0;
        System.out.println( "How many conversions would you like?");
        convNum = sc.nextInt();
        while( convNum < 0 )
        {
            sc.nextLine();
            System.out.println( "Please only enter a positive number");
            convNum = sc.nextInt();
        }


        while( convNum > 0 )
        {
            System.out.println( "inches(1) or cm(2)?");
            int choice = sc.nextInt();
            while( choice < 1 || choice > 2 )
            {
                sc.nextLine();
                System.out.println( "Please only enter a between 1 and 2");
                choice = sc.nextInt();
            }
            boolean isInches;
            if( choice == 1 )
            {
                isInches = true;
            }
            else
            {
                isInches = false;
            }
            
            double input = getUserDouble(isInches);
            boolean isNegative;

            if( input < 0 )
            {
                isNegative = true;
            }
            else
            {
                isNegative = false;
            }

            if(isNegative)
            {
                System.out.println("Please only enter a positive number");
            }
            else
            {
                if( isInches )
                {
                    System.out.println( "Conversion is: " +  inchesToMeters(input) );  
                }
                else
                {
                    System.out.println( "Conversion is: " +  cmTometersOrInches(input, isInches) );  
                }
                
                addToCount();
                --convNum;   
            }
        }


       System.out.println( "Completed " + count + " conversions");
       sc.close();


   }


   public static double getUserDouble( boolean isInches )
   {
       double input;
       if( isInches )
       {
           System.out.println( "Please enter your inches to convert" );
           input = sc.nextDouble();
           while( input < 0 )
           {
               sc.nextLine();
               System.out.println( "Please only enter a positive number");
               input = sc.nextDouble();
           }
       }
       else
       {
           System.out.println( "Please enter your cm to convert" );
           input = sc.nextDouble();
           while( input < 0 )
           {
               sc.nextLine();
               System.out.println( "Please only enter a positive number");
               input = sc.nextDouble();
           }
       }
       return input;
   }




   public static void addToCount()
   {
       ++count;
   }




   public static double inchesToMeters( double inches )
   {
       if( inches > 0 || inches <= 0 )
       {
           System.out.println( "Converting now" );
       }
       ++count;
       --convNum;
       return inches/39.37;
   }




   public static double cmTometersOrInches( double cm, boolean toInches )
   {
       double output;
       if( toInches )
       {
           output = 39.37f*cm;
       }
       else
       {
           output = cm/100;
       }
       ++count;
       --convNum;
       return output;
   }
}



